Daterange and date picker filters

ryanmitchell opened the pull request:
First stab at this, let me know what you think.

sampoyigi commented on app/admin/widgets/filter/assets/js/daterange.js:
```diff
@@ -0,0 +1,104 @@
+    // FIELD CHART CONTROL CLASS DEFINITION
+    // ============================
+
+    var DaterangeControl = function (element, options) {
```
Should be DateRangeControl instead.

ryanmitchell replied:
I've updated it

sampoyigi commented on app/admin/widgets/Filter.php:
```diff
@@ -416,6 +437,27 @@ public function applyScopeToQuery($scope, $query)
                 break;
+                
+            case 'daterange':
+                $value = explode('|', $scope->value);
```
Can you use ` - ` separator instead to match the same format expected by the daterangepicker.js plugin.

ryanmitchell commented:
I didn’t use “-“ as it appears in a date string (Eg 2020-06-06). I had to use a delimiter that doesn’t.

> On 18 Jun 2020, at 15:33, Sam Poyigi <notifications@github.com> wrote:
> 
> ﻿
> @sampoyigi commented on this pull request.
> 
> In app/admin/widgets/Filter.php:
> 
> > @@ -416,6 +437,27 @@ public function applyScopeToQuery($scope, $query)
>                  }
>  
>                  break;
> +                
> +            case 'daterange':
> +                $value = explode('|', $scope->value);
> Can you use - separator instead to match the same format expected by the daterangepicker.js plugin.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

sampoyigi commented:
> I didn’t use “-“ as it appears in a date string (Eg 2020-06-06). I had to use a delimiter that doesn’t.

Ahh, how about using regex to get the start and end date?

ryanmitchell commented:
Yeah could do. I just thought using a different delimiter was cleaner and simpler. Don’t mind changing it if you feel strongly about it.

> On 18 Jun 2020, at 17:44, Sam Poyigi <notifications@github.com> wrote:
> 
> ﻿
> I didn’t use “-“ as it appears in a date string (Eg 2020-06-06). I had to use a delimiter that doesn’t.
> 
> Ahh, how about using regex to get the start and end date?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

sampoyigi commented:
Yea but not strongly, maybe a different approach. | feel passing the correct format to the input value will address the date range selection issue.

ryanmitchell commented:
No problem - easily changed. I’ve updated it now and fixed up the drop down issue too.
Also made datepicker use the same plugin.


> On 18 Jun 2020, at 19:26, Sam Poyigi <notifications@github.com> wrote:
> 
> 
> Yea but not strongly, maybe a different approach. | feel passing the correct format to the input value will address the date range selection issue.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/tastyigniter/TastyIgniter/pull/405#issuecomment-646232513>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAMVO7R7ADMQZCFK2ZY2QLRXJL5LANCNFSM4N3I37KQ>.
>

sampoyigi commented:
Great on fixing the date range issue.

If you check the datepicker on the admin reservations page, the dates are displayed as `NaN` and the page reloads every time the picker is toggled when no date is selected.

<img width="314" alt="Screenshot 2020-06-21 at 13 49 29" src="https://user-images.githubusercontent.com/6567634/85225155-03182800-b3c7-11ea-876a-3f23c1c23ede.png">

ryanmitchell commented:
That should be fixed with the last push - the input-group addition caused issues.



> On 21 Jun 2020, at 13:58, Sam Poyigi <notifications@github.com> wrote:
> 
> 
> Great on fixing the date range issue.
> 
> If you check the datepicker on the admin reservations page, the dates are displayed as NaN and the page reloads every time the picker is toggled when no date is selected.
> 
>  <https://user-images.githubusercontent.com/6567634/85225155-03182800-b3c7-11ea-876a-3f23c1c23ede.png>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/tastyigniter/TastyIgniter/pull/405#issuecomment-647125003>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAMVO3POA63VJMUWEGWEG3RXX7YTANCNFSM4N3I37KQ>.
>

sampoyigi commented:
Still the same issue, I don't think its the input-group as the issue was there before I added the input-group. Does it have anything to do with the picker's expected date format?

ryanmitchell commented:
Very strange - not seeing it on my side and tested on a couple of clean browser sessions to be sure. Are you seeing any JS errors? What is the initial date value being set in the HTML?

I've pushed a check on the initial date to ensure that an invalid date goes back to the default.

sampoyigi commented:
Here's the console log errors, might be related. The initial value set is `21-06-2020` which am guessing is the wrong format.

> Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged and will be removed in an upcoming major release. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.

ryanmitchell commented:
Yeah it should be YYYY-MM-DD format. I’ve pushed a change which should hopefully fix it.

> On 21 Jun 2020, at 22:00, Sam Poyigi <notifications@github.com> wrote:
> 
> 
> Here's the console log errors, might be related. The initial value set is 21-06-2020 which am guessing is the wrong format.
> 
> Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged and will be removed in an upcoming major release. Please refer to http://momentjs.com/guides/#/warnings/js-date/ <http://momentjs.com/guides/#/warnings/js-date/> for more info.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/tastyigniter/TastyIgniter/pull/405#issuecomment-647180990>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAMVO2E3IQJDOKSH42RSRDRXZYGXANCNFSM4N3I37KQ>.
>

sampoyigi commented:
Yep that fixed the issue, I've pushed some improvements, let me know what you think before I merge 👍

ryanmitchell commented:
Yep took a look, happy with what you've done
